### DISCUSSION TITLE
Decoding a NMEA string from a tcp port

### DISCUSSION THREAD

**Discussion by Skyhikeeper** [dotMorten/NmeaParser#90] [General]
I have tried using the parser but not sure why it is not working, I have a string that I receive via a TcpListner and want to decode it, Two questions: can I use the parser to receive the message on a port so not needing to do it via using the TcpListner, and the second how doo i use the parser with a string to decode the string to message data.

Terence

**Comment by Skyhikeeper** [comment]
Using standard .net framework 4.6.2 on windows console application

**Comment by dotMorten** [comment]
>  can I use the parser to receive the message on 

Yes.

>  how do I use the parser with a string to decode the string to message data.

You need a stream. Don't you already get data via the port / listener as a stream?

Alternatively, you could just use this method if you have a single NMEA message:
https://dotmorten.github.io/NmeaParser/api/netfx/NmeaParser.Messages.NmeaMessage.Parse.html#NmeaParser_Messages_NmeaMessage_Parse_System_String_NmeaParser_IMultiSentenceMessage_System_Boolean_
